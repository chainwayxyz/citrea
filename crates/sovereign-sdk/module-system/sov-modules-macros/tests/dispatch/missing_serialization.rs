use sov_modules_api::DispatchCall;

#[derive(DispatchCall)]
struct TestStruct {}

#[derive(DispatchCall)]
#[serialization(Serialize, SomethingElse)]
struct TestStruct2 {}

#[derive(DispatchCall)]
#[serialization(OnlySomethingElse)]
struct TestStruct3 {}

#[derive(DispatchCall)]
#[serialization(Serialize, Deserialize, TryToInjectSomethingForbidden)]
struct TestStruct4 {}

fn main() {}
