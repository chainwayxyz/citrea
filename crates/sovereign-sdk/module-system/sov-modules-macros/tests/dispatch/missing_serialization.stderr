error: Serialization attributes must contain both 'Serialize' and 'Deserialize', but contains '[]'
 --> tests/dispatch/missing_serialization.rs:3:10
  |
3 | #[derive(DispatchCall)]
  |          ^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `DispatchCall` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Serialization attributes must contain both 'Serialize' and 'Deserialize', but contains '["Serialize", "SomethingElse"]'
 --> tests/dispatch/missing_serialization.rs:6:10
  |
6 | #[derive(DispatchCall)]
  |          ^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `DispatchCall` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Serialization attributes must contain both 'Serialize' and 'Deserialize', but contains '["OnlySomethingElse"]'
  --> tests/dispatch/missing_serialization.rs:10:10
   |
10 | #[derive(DispatchCall)]
   |          ^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `DispatchCall` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Serialization attributes can not contain attributes that are not 'Serialize' and 'Deserialize', but contains: '["TryToInjectSomethingForbidden"]'
  --> tests/dispatch/missing_serialization.rs:14:10
   |
14 | #[derive(DispatchCall)]
   |          ^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `DispatchCall` (in Nightly builds, run with -Z macro-backtrace for more info)
